src/main/java/net/countercraft/movecraft/warfare/features/siege/SiegeLeaderListener.java

```diff
@@ -0,0 +1,150 @@
+package net.countercraft.movecraft.warfare.features.siege;
+
+import net.countercraft.movecraft.MovecraftLocation;
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.craft.PlayerCraft;
+import net.countercraft.movecraft.craft.type.CraftType;
+import net.countercraft.movecraft.events.CraftPreTranslateEvent;
+import net.countercraft.movecraft.events.CraftReleaseEvent;
+import net.countercraft.movecraft.events.CraftRotateEvent;
+import net.countercraft.movecraft.events.CraftSinkEvent;
+import net.countercraft.movecraft.util.ChatUtils;
+import net.countercraft.movecraft.warfare.MovecraftWarfare;
+import net.countercraft.movecraft.warfare.config.Config;
+import net.countercraft.movecraft.warfare.features.siege.events.SiegeBroadcastEvent;
+import net.countercraft.movecraft.warfare.localisation.I18nSupport;
+import net.countercraft.movecraft.worldguard.MovecraftWorldGuard;
+import net.countercraft.movecraft.worldguard.utils.IsInRegion;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+
+
+public class SiegeLeaderListener implements Listener {
+    @EventHandler(priority = EventPriority.NORMAL)
+    public void onCraftMovement(CraftPreTranslateEvent event) {
+        Craft leaderCraft = event.getCraft();
+        if (!(leaderCraft instanceof PlayerCraft && Config.SiegeEnable && !event.isCancelled())) return;
```

Comment:
This logic is correct, but I think it'd make sense to simplify it to be if !playercraft || ~siegeenable || cancelled for readability.